src/Input/InputParamTrait.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli\Input;
+
+use ArrayObject;
+use InvalidArgumentException;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+use function sprintf;
+
+trait InputParamTrait
+{
+    /**
+     * @internal
+     * @var null|ArrayObject<string, InputParam>
+     */
+    private $inputParams;
+
+    /**
+     * @param null|mixed $default
+     * @return $this
+     */
+    final public function addParam(
+        string $name,
+        string $description,
+        string $type,
+        bool $required = false,
+        $default = null,
+        array $options = []
+    ) : self {
+        $mode = $type === InputParam::TYPE_BOOL
+            ? InputOption::VALUE_NONE
+            : InputOption::VALUE_REQUIRED;
+
+        $this->addOption(
+            $name,
+            null,
+            $mode,
+            $description
+            // default null, on purpose
+        );
+
+        if ($this->inputParams === null) {
+            $this->inputParams = new ArrayObject();
+        }
+
+        $this->inputParams->offsetSet($name, new InputParam($name, $description, $type, $required, $default, $options));
+
+        return $this;
+    }
+
+    /**
+     * @return null|bool|int|string
+     * @throws InvalidArgumentException
+     */
+    final public function getParam(string $name)
+    {
+        if ($this->inputParams === null || ! $this->inputParams->offsetExists($name)) {
+            throw new InvalidArgumentException(sprintf('Invalid parameter name: %s', $name));
+        }
+
+        /** @var InputInterface $input */
+        $input = $this->getApplication()->getInput();
```

Comment:
Since we're depending on this method, we should likely have an abstract declaration for it:

```php
abstract public function getApplication();
```

which will ensure that if the class composing the trait does not define it, it will be considered invalid.

Reply:
Note that I did not provide a _return value_ - that keeps it compatible with both implementations. 

The main point is that we need to ensure that the method is defined by the class composing the trait.